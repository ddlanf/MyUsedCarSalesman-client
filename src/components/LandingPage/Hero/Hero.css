.hero{
    position: relative;
    margin-top: 15%;
    height: 48vw;
    width: 100%;
    margin-right: auto;
    margin-left: auto;
    background-color: none !important;
}

.landing-page-get-started-link-mobile{
    text-align: center;
    margin: auto;
    padding: 2%;
    position: absolute;
    left: 40%;
    width: 20%;
    text-align: center;
    font-size: 2.5vw;
    background-color: #886060;
    border-radius: 30%;
    border: 1px solid black;
    z-index: 8;
}

.landing-page-get-started-link{
    display: none;
}

.landing-page-get-started-link:hover{
    color: #886060;
    background-color: yellow;
}

.salesman{
    max-height: 550px;
    height: 45vw;
    position: absolute;
    bottom: 0;
    z-index: 3;
    left: 20%;
}

.yellow-car{
    max-height: 400px;
    height: 30vw;
    z-index: 2;
    position: absolute;
    bottom: 0;
    right: 20%;
    animation-name: car-drives-in;
    animation-duration: 3s;
    animation-timing-function: ease-out;
}

@keyframes car-drives-in {
    from{ right: -60%; }
    to { right: 20%; }
}

@keyframes emerge {
    from { bottom: 8%; }
    to { bottom: 80%; opacity: 1; }
}

@media screen and (min-width: 1000px) {
  
    .hero{
        margin-top: 10%;
    }

    .landing-page-get-started-link-mobile{
        display: none;
    }

    .landing-page-get-started-link{
        display:  block;
        text-align: center;
        margin: auto;
        padding: 2%;
        z-index: 1;
        position: absolute;
        opacity: 0;
        right: 40%;
        width: 20%;
        font-size: 2.5vw;
        background-color: #886060;
        border-radius: 30%;
        animation-name: emerge;
        animation-duration: 3s;
        animation-delay: 3s;
        animation-fill-mode: forwards;
        transition: 1s;
        font-family: 'Merriweather Sans', sans-serif;
        border: 1px solid black;
    }
}