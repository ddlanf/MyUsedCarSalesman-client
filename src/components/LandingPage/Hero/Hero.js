import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './Hero.css'

export default class Hero extends Component {
    render() {
        return (
            <div className="hero">
                 <Link
                    className="landing-page-get-started-link-mobile"
                    to='./view-posts'
                    >
                    Get Started
                 </Link>
                 
                 <Link
                    className="landing-page-get-started-link"
                    to='./view-posts'
                    >
                    Get Started
                 </Link>
                 <img 
                        className="salesman"
                        src={require('../../../Utils/myusecarsalesman_pics/repairman.png')}
                        alt="salesman"
                />
                 <img 
                        className="yellow-car"
                        src={require('../../../Utils/myusecarsalesman_pics/car-303614_1280_0.png')}
                        alt="yellow-car"
                />
            </div>
        )
    }
}
