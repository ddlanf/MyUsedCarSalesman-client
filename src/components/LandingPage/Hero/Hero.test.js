import React from 'react';
import ReactDOM from 'react-dom';
import Hero from './Hero';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('Hero component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <Hero/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <Hero/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});