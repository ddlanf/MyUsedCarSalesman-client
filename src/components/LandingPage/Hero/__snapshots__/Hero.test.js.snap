// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders the UI as expected 1`] = `
<div
  className="hero"
>
  <a
    className="landing-page-get-started-link-mobile"
    href="/view-posts"
    onClick={[Function]}
  >
    Get Started
  </a>
  <a
    className="landing-page-get-started-link"
    href="/view-posts"
    onClick={[Function]}
  >
    Get Started
  </a>
  <img
    alt="salesman"
    className="salesman"
    src="salesman.png"
  />
  <img
    alt="yellow-car"
    className="yellow-car"
    src="yellowcar.png"
  />
</div>
`;
